pkg/reconciler/git/clone.go

```diff
@@ -111,14 +112,14 @@ func (r *Cloner) buildAuth() (transport.AuthMethod, error) {
 	}
 
 	secret, err := r.inClusterClientSet.CoreV1().
-		Secrets(r.repo.TokenNamespace).
+		Secrets(tokenNamespace).
 		Get(context.Background(), secretKey, v1.GetOptions{})
 
-	if err != nil && !apierrors.IsNotFound(err) {
+	if err != nil && !apierrors.IsNotFound(err) && !apierrors.IsForbidden(err) {
 		return nil, err
 	}
```

Comment:
This will hide failures in our SKR deployment. If a token-Namespace is defined but the corresponding secret is missing or cannot be read caused by insufficient permissions we should also fail as it's an invalid state, shoudn't we?

Currently, I would assume the GIt request would just return "forbidden" (because the authentication wasn't provided). Could become a challenge to figure whether the credential is wrong  or just not provided by because the secret wasn't readable.

Reply:
I have changed the logic so that token namespace defaults to a "default" namespace. I am lacking the possibility to log anything in clone.go. I suggest adding a logger and informing about that specific case.

Reply:
sure, if we need a logger let's add one